scripts/tests/chiptest/runner.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import subprocess
+import sys
+import threading
+import time
+import pty
+
+from dataclasses import dataclass
+
+
+class LogPipe(threading.Thread):
+
+    def __init__(self, level, capture_delegate=None, name=None):
+        """Setup the object with a logger and a loglevel
+
+            and start the thread
+            """
+        threading.Thread.__init__(self)
+
+        self.daemon = False
+        self.level = level
+        if sys.platform == 'darwin':
+            self.fd_read, self.fd_write = pty.openpty()
+        else:
+            self.fd_read, self.fd_write = os.pipe()
+
+        self.pipeReader = os.fdopen(self.fd_read)
+        self.captured_logs = []
+        self.capture_delegate = capture_delegate
+        self.name = name
+
+        self.start()
+
+    def CapturedLogContains(self, txt: str):
+        for l in self.captured_logs:
+            if txt in l:
+                return True
+        return False
```

Comment:
I just wondering if this works
```suggestion
        return any(txt in l for l in self.captured_logs)
```

Reply:
I think it may but unsure if more readable. Will apply the suggestion since less lines of code.